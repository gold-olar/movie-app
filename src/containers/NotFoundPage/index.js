import React from "react";
import PropTypes from "prop-types";
import NotFound from "../../components/NotFound";

const NotFoundPage = ({ history, type }) => {
  return (
    <>
      <NotFound type={type} history={history} />
    </>
  );
};

NotFoundPage.propTypes = {
  history: PropTypes.shape({
    goBack: PropTypes.func,
  }),
};

export default NotFoundPage;
