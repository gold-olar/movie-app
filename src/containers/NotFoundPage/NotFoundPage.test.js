import React from "react";
import { unmountComponentAtNode } from "react-dom";
import ReactDOM from "react-dom";
import NotFoundPage from "./index";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});
const testComponent = (
  <BrowserRouter>
    <NotFoundPage />
  </BrowserRouter>
);

test("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(testComponent, div);
});

test("should match snapshot", () => {
  const component = renderer.create(testComponent);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
