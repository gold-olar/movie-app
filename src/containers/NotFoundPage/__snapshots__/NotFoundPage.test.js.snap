// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="mt-5 container-fluid"
>
  <div
    className="mb-5 justify-content-center row"
  >
    <div
      className="text-center col-md-6"
    >
      <h3
        className="appology-msg mb-3"
      >
        We are sorry, the page you are looking for does not exist
      </h3>
      <img
        alt="page not found"
        className="img-fluid mt-3 mb-3"
        src="not_found.svg"
      />
      <h3
        className="not-found-text mt-3 mb-5"
      >
         Not Found 
      </h3>
      <button
        className="btn-primary btn mr-3"
        onClick={[Function]}
      >
        Go Back
      </button>
      <a
        className="secondary-option"
        href="/"
        onClick={[Function]}
      >
        Go Home
      </a>
    </div>
  </div>
</div>
`;
